.travis.yml

```diff
@@ -1,16 +1,81 @@
-sudo: required
 language: c
 
-os:
-  - linux
-  - osx
+sudo: false
 
-compiler:
-  - clang
-  - gcc
+cache:
+  directories:
+    - $HOME/archives
+
+matrix:
+  include:
+  - os: linux
+    compiler: gcc
+    env:
+      TOOLCHAIN=NATIVE
+      TARGET=native
+  - os: linux
+    compiler: clang
+    env:
+      TOOLCHAIN=NATIVE
+      TARGET=native
+  - os: osx
+    compiler: gcc
+    env:
+      TOOLCHAIN=NATIVE
+      TARGET=native
+  - os: osx
+    compiler: clang
+    env:
+      TOOLCHAIN=NATIVE
+      TARGET=native
+  # Toolchains for little-endian, 64-bit ARMv8 for GNU/Linux systems
+  - os: linux
+    env:
+      TOOLCHAIN=LINARO
+      TARGET=aarch64-linux-gnu
+      QEMU_ARCH=aarch64
+  # Toolchains for little-endian, hard-float, 32-bit ARMv7 (and earlier) for GNU/Linux systems
+  - os: linux
+    env:
+      TOOLCHAIN=LINARO
+      TARGET=arm-linux-gnueabihf
+      QEMU_ARCH=arm
+  # Toolchains for little-endian, 32-bit ARMv8 for GNU/Linux systems
+  - os: linux
+    env:
+      TOOLCHAIN=LINARO
+      TARGET=armv8l-linux-gnueabihf
+      QEMU_ARCH=arm
+  # Toolchains for little-endian, soft-float, 32-bit ARMv7 (and earlier) for GNU/Linux systems
+  - os: linux
+    env:
+      TOOLCHAIN=LINARO
+      TARGET=arm-linux-gnueabi
+      QEMU_ARCH=arm
+  # Toolchains for big-endian, 64-bit ARMv8 for GNU/Linux systems
+  - os: linux
+    env:
+      TOOLCHAIN=LINARO
+      TARGET=aarch64_be-linux-gnu
+      QEMU_ARCH=DISABLED
+  # Toolchains for big-endian, hard-float, 32-bit ARMv7 (and earlier) for GNU/Linux systems
+  - os: linux
+    env:
+      TOOLCHAIN=LINARO
+      TARGET=armeb-linux-gnueabihf
+      QEMU_ARCH=DISABLED
+  # Toolchains for big-endian, soft-float, 32-bit ARMv7 (and earlier) for GNU/Linux systems
+  - os: linux
+    env:
+      TOOLCHAIN=LINARO
+      TARGET=armeb-linux-gnueabi
+      QEMU_ARCH=DISABLED
+  - os: linux
+    env:
+      TOOLCHAIN=CODESCAPE
+      TARGET=mips-mti-linux-gnu
+      QEMU_ARCH=DISABLED
 
 script:
   - cmake --version
-  - cmake -DCMAKE_BUILD_TYPE=Debug -DBUILD_TESTING=ON -H. -Bcmake_build
-  - cmake --build cmake_build --target all
-  - CTEST_OUTPUT_ON_FAILURE=1 cmake --build cmake_build --target test
+  - bash -e -x ./scripts/run_integration.sh
```

Comment:
if you look at your log 75% seems to be cross toolchain/qemu build
-> could you create an "install" script and use the "install:" key instead ?
Since travis only provides "folding" for each build steps....

Otherwise need to scroll at bottom to see the cmake (useful) traces...

Reply:
Good idea. I'll do it later though.